merkletree/merkletreekv_test.go

```diff
@@ -0,0 +1,128 @@
+package merkletree
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/coniks-sys/coniks-go/storage/kv"
+	"github.com/coniks-sys/coniks-go/utils"
+)
+
+func TestTreeStore(t *testing.T) {
+	util.WithDB(func(db kv.DB) {
+		key1 := "key1"
+		val1 := []byte("value1")
+		key2 := "key2"
+		val2 := []byte("value2")
+
+		m1, err := NewMerkleTree()
+		if err != nil {
+			t.Fatal(err)
+		}
+		index1 := vrfPrivKey1.Compute([]byte(key1))
+		if err := m1.Set(index1, key1, val1); err != nil {
+			t.Fatal(err)
+		}
+		index2 := vrfPrivKey1.Compute([]byte(key2))
+		if err := m1.Set(index2, key2, val2); err != nil {
+			t.Fatal(err)
+		}
+		m1.recomputeHash()
+
+		wb := db.NewBatch()
+		m1.StoreToKV(1, wb)
+		err = db.Write(wb)
```

Comment:
Is there a reason why ``db.Write(wb)`` isn't called from within ``m.StoreToKV()``?